spring-session-data-redis/src/main/java/org/springframework/session/data/redis/ReactiveRedisOperationsSessionRepository.java

```diff
@@ -192,6 +198,24 @@ private String getSessionKey(String sessionId) {
 		return this.namespace + "sessions:" + sessionId;
 	}
 
+	private String getPrincipalKey(String principalName) {
+		return this.namespace + "index:"
+				+ FindByIndexNameSessionRepository.PRINCIPAL_NAME_INDEX_NAME + ":"
+				+ principalName;
+	}
+
+	@Override
+	public Flux<Tuple2<String, RedisSession>> findByIndexNameAndIndexValue(String indexName, String indexValue) {
+		if (!PRINCIPAL_NAME_INDEX_NAME.equals(indexName)) {
+			return Flux.empty();
+		}
+		String principalKey = getPrincipalKey(indexValue);
+		return this.sessionRedisOperations.opsForSet()
+				.scan(principalKey)
```

Comment:
I'm afraid this isn't optimal, as you would be scanning for a match each time, rather having an index that's updated on write and have reads simple.